{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# Tandem Repeat analysis\n",
    "#### Written by Jigar N. Bandaria\n",
    "In this notebook I will confirm my analysis. I only want to save sgRNA sequences that have NGG at the 3' end. The sequences should also be repetitive atleast 4 times within a 10 kbp window, and be present only on one chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here I use Bowtie to first align all the sequences that we had saved from the previous notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Time loading forward index: 00:00:09\n",
      "Time for 0-mismatch search: 00:00:06\n",
      "# reads processed: 70795\n",
      "# reads with at least one reported alignment: 70795 (100.00%)\n",
      "# reads that failed to align: 0 (0.00%)\n",
      "Reported 729062 alignments to 1 output stream(s)\n",
      "Time searching: 00:00:15\n",
      "Overall time: 00:00:15\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/home/user/Desktop/Bioinformatics/bowtie-1.1.2/./bowtie -t -S -a -v 0 -y -f  /home/user/Desktop/Bioinformatics/genome_index/GRCh37 /home/user/Desktop/nt_17/no_rev_comp.fasta > no_rev_comp_grc37.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112957261</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112941860</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112934077</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112963854</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112954065</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqid  strand    chr      start           Sequence\n",
       "0  seq0       0  chr13  112957261  GGACCATTCCTTCAGGA\n",
       "1  seq0       0  chr13  112941860  GGACCATTCCTTCAGGA\n",
       "2  seq0       0  chr13  112934077  GGACCATTCCTTCAGGA\n",
       "3  seq0       0  chr13  112963854  GGACCATTCCTTCAGGA\n",
       "4  seq0       0  chr13  112954065  GGACCATTCCTTCAGGA"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#above I have checked my sequences in human genome. I am loading the results.\n",
    "unmatched_seq = pd.read_table('/home/user/Desktop/nt_17/no_rev_comp_grc37.sam',skiprows=27,header=None,usecols=[0,1,2,3,9], names=['seqid','strand','chr','start','Sequence'])\n",
    "unmatched_seq.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to confirm that the sequences are follows by NGG or preceded by CCN (if they are the opposite strand) I used bedtools to slop those sequences. This is all included in the bash script 'compare_ngg.sh'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "./compare_ngg.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr13:112957257-112957280(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr13:112941856-112941879(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr13:112934073-112934096(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr13:112963850-112963873(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr13:112954061-112954084(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          chr_p                    seq23\n",
       "0  chr13:112957257-112957280(+)  CATGGACCATTCCTTCAGGATGG\n",
       "1  chr13:112941856-112941879(+)  CATGGACCATTCCTTCAGGATGG\n",
       "2  chr13:112934073-112934096(+)  CATGGACCATTCCTTCAGGATGG\n",
       "3  chr13:112963850-112963873(+)  CATGGACCATTCCTTCAGGATGG\n",
       "4  chr13:112954061-112954084(+)  CATGGACCATTCCTTCAGGATGG"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#loading the slopped sequences:\n",
    "slopped_seq = pd.read_table('/home/user/Desktop/nt_17/slop_fasta.fa',header=None,names=['chr_p','seq23'])\n",
    "slopped_seq['seq23'] = slopped_seq.seq23.str.upper()\n",
    "slopped_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "729062 729062\n"
     ]
    }
   ],
   "source": [
    "#check length of unmatched_seq and slopped_seq. Are they equal?\n",
    "print (len(unmatched_seq),len(slopped_seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "      <th>NGG5</th>\n",
       "      <th>NGG3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>729052</th>\n",
       "      <td>chr10:37467845-37467868(+)</td>\n",
       "      <td>GTGGGTGCTCTGGAGACTACTGG</td>\n",
       "      <td>GTG</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729053</th>\n",
       "      <td>chr21:18929000-18929023(-)</td>\n",
       "      <td>CCAGGTGCTCTGGAGACTACATC</td>\n",
       "      <td>CCA</td>\n",
       "      <td>ATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729054</th>\n",
       "      <td>chr17:80108900-80108923(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729055</th>\n",
       "      <td>chr17:80107712-80107735(-)</td>\n",
       "      <td>TCCGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729056</th>\n",
       "      <td>chr17:80107748-80107771(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729057</th>\n",
       "      <td>chr17:80109350-80109373(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729058</th>\n",
       "      <td>chr10:393010-393033(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729059</th>\n",
       "      <td>chr10:393417-393440(-)</td>\n",
       "      <td>GCAATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCA</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729060</th>\n",
       "      <td>chr10:393096-393119(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729061</th>\n",
       "      <td>chr10:393053-393076(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             chr_p                    seq23 NGG5 NGG3\n",
       "729052  chr10:37467845-37467868(+)  GTGGGTGCTCTGGAGACTACTGG  GTG  TGG\n",
       "729053  chr21:18929000-18929023(-)  CCAGGTGCTCTGGAGACTACATC  CCA  ATC\n",
       "729054  chr17:80108900-80108923(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG\n",
       "729055  chr17:80107712-80107735(-)  TCCGGGTGTGCGCCAGGTTCTGG  TCC  TGG\n",
       "729056  chr17:80107748-80107771(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG\n",
       "729057  chr17:80109350-80109373(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG\n",
       "729058      chr10:393010-393033(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG\n",
       "729059      chr10:393417-393440(-)  GCAATGTGTCCGGGTGAGTGTGG  GCA  TGG\n",
       "729060      chr10:393096-393119(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG\n",
       "729061      chr10:393053-393076(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Adding the 3 nucleotides that are PAMs\n",
    "slopped_seq['NGG5'] = slopped_seq['seq23'].str[:3]\n",
    "slopped_seq['NGG3'] = slopped_seq['seq23'].str[-3:]\n",
    "slopped_seq.tail(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "      <th>NGG5</th>\n",
       "      <th>NGG3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112957261</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "      <td>chr13:112957257-112957280(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "      <td>CAT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112941860</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "      <td>chr13:112941856-112941879(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "      <td>CAT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112934077</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "      <td>chr13:112934073-112934096(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "      <td>CAT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112963854</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "      <td>chr13:112963850-112963873(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "      <td>CAT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>seq0</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112954065</td>\n",
       "      <td>GGACCATTCCTTCAGGA</td>\n",
       "      <td>chr13:112954061-112954084(+)</td>\n",
       "      <td>CATGGACCATTCCTTCAGGATGG</td>\n",
       "      <td>CAT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqid  strand    chr      start           Sequence  \\\n",
       "0  seq0       0  chr13  112957261  GGACCATTCCTTCAGGA   \n",
       "1  seq0       0  chr13  112941860  GGACCATTCCTTCAGGA   \n",
       "2  seq0       0  chr13  112934077  GGACCATTCCTTCAGGA   \n",
       "3  seq0       0  chr13  112963854  GGACCATTCCTTCAGGA   \n",
       "4  seq0       0  chr13  112954065  GGACCATTCCTTCAGGA   \n",
       "\n",
       "                          chr_p                    seq23 NGG5 NGG3  \n",
       "0  chr13:112957257-112957280(+)  CATGGACCATTCCTTCAGGATGG  CAT  TGG  \n",
       "1  chr13:112941856-112941879(+)  CATGGACCATTCCTTCAGGATGG  CAT  TGG  \n",
       "2  chr13:112934073-112934096(+)  CATGGACCATTCCTTCAGGATGG  CAT  TGG  \n",
       "3  chr13:112963850-112963873(+)  CATGGACCATTCCTTCAGGATGG  CAT  TGG  \n",
       "4  chr13:112954061-112954084(+)  CATGGACCATTCCTTCAGGATGG  CAT  TGG  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Concatenating both the file obtained from Bowtie and Bedtools.\n",
    "slop_both = pd.concat([unmatched_seq,slopped_seq],axis=1)\n",
    "slop_both.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I will perform analysis to make sure that the sequences that present only on one chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "seqid\n",
       "seq0                                  [chr13]\n",
       "seq1                                   [chr8]\n",
       "seq10                           [chr3, chr15]\n",
       "seq100                     [chr3, chr6, chr4]\n",
       "seq1000                               [chr17]\n",
       "seq10000                              [chr17]\n",
       "seq10001                               [chr4]\n",
       "seq10002                 [chr11, chr13, chr1]\n",
       "seq10003    [chr17, chr10, chr1, chr18, chr4]\n",
       "seq10004                               [chr5]\n",
       "seq10005                               [chr3]\n",
       "seq10006                        [chr9, chr22]\n",
       "seq10007                        [chr19, chr2]\n",
       "seq10008                              [chr21]\n",
       "seq10009                [chr21, chr19, chr16]\n",
       "seq1001                                [chr4]\n",
       "seq10010                              [chr12]\n",
       "seq10011                              [chr13]\n",
       "seq10012                               [chr2]\n",
       "seq10013                              [chr10]\n",
       "seq10014                              [chr19]\n",
       "seq10015                               [chr2]\n",
       "seq10016                              [chr17]\n",
       "seq10017                        [chr2, chr19]\n",
       "seq10018            [chr19, chrX, chr2, chr6]\n",
       "seq10019                               [chr9]\n",
       "seq1002                                [chr1]\n",
       "seq10020                              [chr10]\n",
       "seq10021                               [chr9]\n",
       "seq10022                       [chr16, chr14]\n",
       "                          ...                \n",
       "seq9972                        [chr10, chr22]\n",
       "seq9973                                [chr2]\n",
       "seq9974                                [chr6]\n",
       "seq9975                  [chr22, chr2, chr14]\n",
       "seq9976                                [chr5]\n",
       "seq9977                               [chr22]\n",
       "seq9978                                [chrX]\n",
       "seq9979                               [chr17]\n",
       "seq998              [chr2, chr4, chr15, chr6]\n",
       "seq9980                               [chr14]\n",
       "seq9981                                [chrX]\n",
       "seq9982                               [chr10]\n",
       "seq9983                               [chr17]\n",
       "seq9984                                [chr7]\n",
       "seq9985                                [chrX]\n",
       "seq9986                                [chr2]\n",
       "seq9987                               [chr20]\n",
       "seq9988                               [chr15]\n",
       "seq9989                                [chr5]\n",
       "seq999                                [chr16]\n",
       "seq9990                                [chr3]\n",
       "seq9991                   [chr10, chrX, chr2]\n",
       "seq9992                               [chr18]\n",
       "seq9993                                [chr1]\n",
       "seq9994                               [chr13]\n",
       "seq9995            [chr11, chr13, chr9, chr7]\n",
       "seq9996                                [chr7]\n",
       "seq9997                                [chr4]\n",
       "seq9998                                [chr7]\n",
       "seq9999                               [chr17]\n",
       "Name: chr, dtype: object"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grp1 = slop_both.groupby('seqid')['chr'].unique()\n",
    "grp1#grouping to check if any sequences on multiple chromosomes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "70795"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(grp1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "grp2 = grp1.value_counts() #counting how many on each chromosome  based on seq_id\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "import pickle\n",
    "with open('grp2.pkl','wb') as fid:\n",
    "    pickle.dump (grp2,fid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[chr1]                                    3755\n",
       "[chr2]                                    3015\n",
       "[chr19]                                   2977\n",
       "[chr7]                                    2966\n",
       "[chr10]                                   2831\n",
       "[chr8]                                    2439\n",
       "[chr17]                                   2196\n",
       "[chr4]                                    2045\n",
       "[chr6]                                    1994\n",
       "[chr13]                                   1966\n",
       "[chr5]                                    1949\n",
       "[chr9]                                    1924\n",
       "[chr16]                                   1814\n",
       "[chr12]                                   1747\n",
       "[chrX]                                    1706\n",
       "[chr11]                                   1640\n",
       "[chr18]                                   1388\n",
       "[chr22]                                   1226\n",
       "[chr20]                                   1207\n",
       "[chr3]                                     997\n",
       "[chr21]                                    879\n",
       "[chr14]                                    720\n",
       "[chr15]                                    585\n",
       "[chrY]                                     361\n",
       "[chr8, chr11]                                1\n",
       "[chr7, chr8]                                 1\n",
       "[chr3, chrX, chr11, chr1]                    1\n",
       "[chr16, chr6]                                1\n",
       "[chr7, chr1]                                 1\n",
       "[chr15, chr16]                               1\n",
       "                                          ... \n",
       "[chr6, chr2, chr11]                          1\n",
       "[chr22, chr10, chr6, chr16, chr17]           1\n",
       "[chr8, chr19]                                1\n",
       "[chr16, chr20, chr8]                         1\n",
       "[chr11, chr8]                                1\n",
       "[chr15, chrY]                                1\n",
       "[chr12, chr1]                                1\n",
       "[chr10, chr6]                                1\n",
       "[chr12, chr2]                                1\n",
       "[chrX, chr4, chr15, chr1, chr6, chr16]       1\n",
       "[chr19, chr7]                                1\n",
       "[chr15, chr7]                                1\n",
       "[chr21, chr5, chr14]                         1\n",
       "[chr20, chr14]                               1\n",
       "[chr5, chr22]                                1\n",
       "[chr3, chr2]                                 1\n",
       "[chr16, chr1]                                1\n",
       "[chr18, chr6]                                1\n",
       "[chrX, chr15, chr12, chr2, chr8]             1\n",
       "[chr5, chr6]                                 1\n",
       "[chr18, chr13]                               1\n",
       "[chr9, chr17]                                1\n",
       "[chr11, chr6]                                1\n",
       "[chr1, chr10]                                1\n",
       "[chr8, chr2]                                 1\n",
       "[chr2, chr20]                                1\n",
       "[chr11, chr14]                               1\n",
       "[chrX, chr9]                                 1\n",
       "[chr2, chr16]                                1\n",
       "[chrX, chr1, chr17]                          1\n",
       "Name: chr, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grp2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are some sequences that are present on multiple chromosome. This is not because of some error in our previous analysis. This is because Bowtie returns all the homologous sequences in the genome, whether they are followed by a PAM or not. Once we remove the sequences from the Bowtie analysis that are not followed by NGG, we would observe that only chromosome specific sequences will remain."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1       26468\n",
       "879         1\n",
       "1207        1\n",
       "2977        1\n",
       "1747        1\n",
       "1924        1\n",
       "2196        1\n",
       "997         1\n",
       "2439        1\n",
       "3015        1\n",
       "1814        1\n",
       "2966        1\n",
       "1640        1\n",
       "1966        1\n",
       "361         1\n",
       "585         1\n",
       "1226        1\n",
       "1706        1\n",
       "3755        1\n",
       "1994        1\n",
       "1388        1\n",
       "1949        1\n",
       "2045        1\n",
       "2831        1\n",
       "720         1\n",
       "Name: chr, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grp2.value_counts()#counting how many ones based on seq_id\n",
    "#These sequences are present on more than one chrosome. Need to check if when on other\n",
    "#chromosomes they have a NGG on 3'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "mask1 = [len(x)!=1 for x in grp1] \n",
    "#creating a mask to get the ones that are on more than one chr."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "more_than_1 = grp1[mask1] #this is the list of more than 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "seqid\n",
       "seq10                           [chr3, chr15]\n",
       "seq100                     [chr3, chr6, chr4]\n",
       "seq10002                 [chr11, chr13, chr1]\n",
       "seq10003    [chr17, chr10, chr1, chr18, chr4]\n",
       "seq10006                        [chr9, chr22]\n",
       "Name: chr, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "more_than_1[:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below is a dataframe showing sequences that are present on more than 1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(330250, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "      <th>NGG5</th>\n",
       "      <th>NGG3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>631513</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152327464</td>\n",
       "      <td>GATGACTGACTTGAGCC</td>\n",
       "      <td>chr1:152327460-152327483(-)</td>\n",
       "      <td>TCTGGCTCAAGTCAGTCATCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631514</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152328376</td>\n",
       "      <td>GATGACTGACTTGAGCC</td>\n",
       "      <td>chr1:152328372-152328395(-)</td>\n",
       "      <td>TCTGGCTCAAGTCAGTCATCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631515</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152327920</td>\n",
       "      <td>GATGACTGACTTGAGCC</td>\n",
       "      <td>chr1:152327916-152327939(-)</td>\n",
       "      <td>TCTGGCTCAAGTCAGTCATCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631517</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>16</td>\n",
       "      <td>chr1</td>\n",
       "      <td>152327233</td>\n",
       "      <td>GATGACTGACTTGAGCC</td>\n",
       "      <td>chr1:152327229-152327252(-)</td>\n",
       "      <td>TCTGGCTCAAGTCAGTCATCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631512</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>0</td>\n",
       "      <td>chr14</td>\n",
       "      <td>65118519</td>\n",
       "      <td>GGCTCAAGTCAGTCATC</td>\n",
       "      <td>chr14:65118515-65118538(+)</td>\n",
       "      <td>TTGGGCTCAAGTCAGTCATCCTC</td>\n",
       "      <td>TTG</td>\n",
       "      <td>CTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631511</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>0</td>\n",
       "      <td>chr19</td>\n",
       "      <td>42218255</td>\n",
       "      <td>GGCTCAAGTCAGTCATC</td>\n",
       "      <td>chr19:42218251-42218274(+)</td>\n",
       "      <td>TCTGGCTCAAGTCAGTCATCATC</td>\n",
       "      <td>TCT</td>\n",
       "      <td>ATC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631516</th>\n",
       "      <td>seq55055</td>\n",
       "      <td>16</td>\n",
       "      <td>chr19</td>\n",
       "      <td>43027546</td>\n",
       "      <td>GATGACTGACTTGAGCC</td>\n",
       "      <td>chr19:43027542-43027565(-)</td>\n",
       "      <td>ACTGGCTCAAGTCAGTCATCATC</td>\n",
       "      <td>ACT</td>\n",
       "      <td>ATC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           seqid  strand    chr      start           Sequence  \\\n",
       "631513  seq55055      16   chr1  152327464  GATGACTGACTTGAGCC   \n",
       "631514  seq55055      16   chr1  152328376  GATGACTGACTTGAGCC   \n",
       "631515  seq55055      16   chr1  152327920  GATGACTGACTTGAGCC   \n",
       "631517  seq55055      16   chr1  152327233  GATGACTGACTTGAGCC   \n",
       "631512  seq55055       0  chr14   65118519  GGCTCAAGTCAGTCATC   \n",
       "631511  seq55055       0  chr19   42218255  GGCTCAAGTCAGTCATC   \n",
       "631516  seq55055      16  chr19   43027546  GATGACTGACTTGAGCC   \n",
       "\n",
       "                              chr_p                    seq23 NGG5 NGG3  \n",
       "631513  chr1:152327460-152327483(-)  TCTGGCTCAAGTCAGTCATCTGG  TCT  TGG  \n",
       "631514  chr1:152328372-152328395(-)  TCTGGCTCAAGTCAGTCATCTGG  TCT  TGG  \n",
       "631515  chr1:152327916-152327939(-)  TCTGGCTCAAGTCAGTCATCTGG  TCT  TGG  \n",
       "631517  chr1:152327229-152327252(-)  TCTGGCTCAAGTCAGTCATCTGG  TCT  TGG  \n",
       "631512   chr14:65118515-65118538(+)  TTGGGCTCAAGTCAGTCATCCTC  TTG  CTC  \n",
       "631511   chr19:42218251-42218274(+)  TCTGGCTCAAGTCAGTCATCATC  TCT  ATC  \n",
       "631516   chr19:43027542-43027565(-)  ACTGGCTCAAGTCAGTCATCATC  ACT  ATC  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create data with only more than one sequences\n",
    "remain_seq_1 = slop_both[slop_both.seqid.isin(more_than_1.index)]\n",
    "print(remain_seq_1.shape)\n",
    "remain_seq_1[remain_seq_1.seqid=='seq55055'].sort_values(['chr'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below are check for the ones that are followed by NGG and remove the ones that are not."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "203532"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#based on some checking the criteria whould be NGG3\n",
    "remain_seq_2 = remain_seq_1[remain_seq_1.NGG3.isin(['AGG','TGG','CGG','GGG'])]\n",
    "len(remain_seq_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "      <th>NGG5</th>\n",
       "      <th>NGG3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>seq3</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112956894</td>\n",
       "      <td>TTCTTCAGGATGGGCCC</td>\n",
       "      <td>chr13:112956890-112956913(+)</td>\n",
       "      <td>TGTTTCTTCAGGATGGGCCCAGG</td>\n",
       "      <td>TGT</td>\n",
       "      <td>AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>seq3</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112964439</td>\n",
       "      <td>TTCTTCAGGATGGGCCC</td>\n",
       "      <td>chr13:112964435-112964458(+)</td>\n",
       "      <td>TGTTTCTTCAGGATGGGCCCAGG</td>\n",
       "      <td>TGT</td>\n",
       "      <td>AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>seq3</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112951863</td>\n",
       "      <td>TTCTTCAGGATGGGCCC</td>\n",
       "      <td>chr13:112951859-112951882(+)</td>\n",
       "      <td>TGTTTCTTCAGGATGGGCCCAGG</td>\n",
       "      <td>TGT</td>\n",
       "      <td>AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>seq3</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112934832</td>\n",
       "      <td>TTCTTCAGGATGGGCCC</td>\n",
       "      <td>chr13:112934828-112934851(+)</td>\n",
       "      <td>TGTTTCTTCAGGATGGGCCCAGG</td>\n",
       "      <td>TGT</td>\n",
       "      <td>AGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>seq3</td>\n",
       "      <td>0</td>\n",
       "      <td>chr13</td>\n",
       "      <td>112962638</td>\n",
       "      <td>TTCTTCAGGATGGGCCC</td>\n",
       "      <td>chr13:112962634-112962657(+)</td>\n",
       "      <td>TGTTTCTTCAGGATGGGCCCAGG</td>\n",
       "      <td>TGT</td>\n",
       "      <td>AGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    seqid  strand    chr      start           Sequence  \\\n",
       "809  seq3       0  chr13  112956894  TTCTTCAGGATGGGCCC   \n",
       "810  seq3       0  chr13  112964439  TTCTTCAGGATGGGCCC   \n",
       "811  seq3       0  chr13  112951863  TTCTTCAGGATGGGCCC   \n",
       "812  seq3       0  chr13  112934832  TTCTTCAGGATGGGCCC   \n",
       "813  seq3       0  chr13  112962638  TTCTTCAGGATGGGCCC   \n",
       "\n",
       "                            chr_p                    seq23 NGG5 NGG3  \n",
       "809  chr13:112956890-112956913(+)  TGTTTCTTCAGGATGGGCCCAGG  TGT  AGG  \n",
       "810  chr13:112964435-112964458(+)  TGTTTCTTCAGGATGGGCCCAGG  TGT  AGG  \n",
       "811  chr13:112951859-112951882(+)  TGTTTCTTCAGGATGGGCCCAGG  TGT  AGG  \n",
       "812  chr13:112934828-112934851(+)  TGTTTCTTCAGGATGGGCCCAGG  TGT  AGG  \n",
       "813  chr13:112962634-112962657(+)  TGTTTCTTCAGGATGGGCCCAGG  TGT  AGG  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "remain_seq_2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rechecking to make sure now that each sequence is present only one chromosome."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "seqid\n",
       "seq10        [chr3]\n",
       "seq100       [chr3]\n",
       "seq10002    [chr11]\n",
       "seq10003    [chr17]\n",
       "seq10006    [chr22]\n",
       "seq10007    [chr19]\n",
       "seq10009    [chr19]\n",
       "seq10017     [chr2]\n",
       "seq10018    [chr19]\n",
       "seq10022    [chr16]\n",
       "Name: chr, dtype: object"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check the groupings now\n",
    "#check1 : by seq_id\n",
    "\n",
    "check_1 = remain_seq_2.groupby('seqid')['chr'].unique()\n",
    "check_1[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[chr19]    2025\n",
       "[chr1]     1953\n",
       "[chr2]     1867\n",
       "[chr10]    1687\n",
       "[chr7]     1674\n",
       "[chr8]     1442\n",
       "[chr17]    1307\n",
       "[chr6]     1213\n",
       "[chr16]    1143\n",
       "[chr4]     1140\n",
       "[chr9]     1137\n",
       "[chr12]    1081\n",
       "[chr5]     1077\n",
       "[chr13]    1033\n",
       "[chr11]     978\n",
       "[chr20]     923\n",
       "[chrX]      853\n",
       "[chr18]     801\n",
       "[chr22]     778\n",
       "[chr3]      613\n",
       "[chr21]     598\n",
       "[chr14]     481\n",
       "[chr15]     369\n",
       "[chrY]      295\n",
       "Name: chr, dtype: int64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "check_1.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "#### Now there are no sequences that are only more than one chromosomes. This means that the sequences that were found on other chromosomes were not followed by PAM sequences. Hence, all the files in the unmatched sequences are fine. \n",
    "\n",
    "Below is the final analysis that will be saved."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initial ALL Alignments : 729062\n",
      "Remianing sequences after removing non-NGG : 554634\n"
     ]
    }
   ],
   "source": [
    "#final analysis on the all the alignments slop_both\n",
    "#ALL aligments also include matches on other chromosomes but dont have NGG at 3'\n",
    "#I want to remove them\n",
    "#first add another column seq23\n",
    "\n",
    "slop_both['seq20'] = slop_both['seq23'].str[-20:]\n",
    "\n",
    "print ('Initial ALL Alignments : {}'.format(len(slop_both)))\n",
    "\n",
    "true_alignments = slop_both[slop_both.NGG3.isin(['AGG','TGG','CGG','GGG'])]\n",
    "print ('Remianing sequences after removing non-NGG : {}'.format(len(true_alignments)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total sequences based on seq_id groupby : 70795\n"
     ]
    }
   ],
   "source": [
    "print('Total sequences based on seq_id groupby : {}'.format(len(true_alignments.groupby('seqid').sum())))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>chr_p</th>\n",
       "      <th>seq23</th>\n",
       "      <th>NGG5</th>\n",
       "      <th>NGG3</th>\n",
       "      <th>seq20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>729051</th>\n",
       "      <td>seq70792</td>\n",
       "      <td>0</td>\n",
       "      <td>chr10</td>\n",
       "      <td>37456123</td>\n",
       "      <td>GGTGCTCTGGAGACTAC</td>\n",
       "      <td>chr10:37456119-37456142(+)</td>\n",
       "      <td>GTGGGTGCTCTGGAGACTACTGG</td>\n",
       "      <td>GTG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGTGCTCTGGAGACTACTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729052</th>\n",
       "      <td>seq70792</td>\n",
       "      <td>0</td>\n",
       "      <td>chr10</td>\n",
       "      <td>37467849</td>\n",
       "      <td>GGTGCTCTGGAGACTAC</td>\n",
       "      <td>chr10:37467845-37467868(+)</td>\n",
       "      <td>GTGGGTGCTCTGGAGACTACTGG</td>\n",
       "      <td>GTG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGTGCTCTGGAGACTACTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729054</th>\n",
       "      <td>seq70793</td>\n",
       "      <td>16</td>\n",
       "      <td>chr17</td>\n",
       "      <td>80108904</td>\n",
       "      <td>GAACCTGGCGCACACCC</td>\n",
       "      <td>chr17:80108900-80108923(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGGTGTGCGCCAGGTTCTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729055</th>\n",
       "      <td>seq70793</td>\n",
       "      <td>16</td>\n",
       "      <td>chr17</td>\n",
       "      <td>80107716</td>\n",
       "      <td>GAACCTGGCGCACACCC</td>\n",
       "      <td>chr17:80107712-80107735(-)</td>\n",
       "      <td>TCCGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCC</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGGTGTGCGCCAGGTTCTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729056</th>\n",
       "      <td>seq70793</td>\n",
       "      <td>16</td>\n",
       "      <td>chr17</td>\n",
       "      <td>80107752</td>\n",
       "      <td>GAACCTGGCGCACACCC</td>\n",
       "      <td>chr17:80107748-80107771(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGGTGTGCGCCAGGTTCTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729057</th>\n",
       "      <td>seq70793</td>\n",
       "      <td>16</td>\n",
       "      <td>chr17</td>\n",
       "      <td>80109354</td>\n",
       "      <td>GAACCTGGCGCACACCC</td>\n",
       "      <td>chr17:80109350-80109373(-)</td>\n",
       "      <td>TCTGGGTGTGCGCCAGGTTCTGG</td>\n",
       "      <td>TCT</td>\n",
       "      <td>TGG</td>\n",
       "      <td>GGGTGTGCGCCAGGTTCTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729058</th>\n",
       "      <td>seq70794</td>\n",
       "      <td>16</td>\n",
       "      <td>chr10</td>\n",
       "      <td>393014</td>\n",
       "      <td>CACTCACCCGGACACAT</td>\n",
       "      <td>chr10:393010-393033(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>ATGTGTCCGGGTGAGTGTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729059</th>\n",
       "      <td>seq70794</td>\n",
       "      <td>16</td>\n",
       "      <td>chr10</td>\n",
       "      <td>393421</td>\n",
       "      <td>CACTCACCCGGACACAT</td>\n",
       "      <td>chr10:393417-393440(-)</td>\n",
       "      <td>GCAATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCA</td>\n",
       "      <td>TGG</td>\n",
       "      <td>ATGTGTCCGGGTGAGTGTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729060</th>\n",
       "      <td>seq70794</td>\n",
       "      <td>16</td>\n",
       "      <td>chr10</td>\n",
       "      <td>393100</td>\n",
       "      <td>CACTCACCCGGACACAT</td>\n",
       "      <td>chr10:393096-393119(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>ATGTGTCCGGGTGAGTGTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>729061</th>\n",
       "      <td>seq70794</td>\n",
       "      <td>16</td>\n",
       "      <td>chr10</td>\n",
       "      <td>393057</td>\n",
       "      <td>CACTCACCCGGACACAT</td>\n",
       "      <td>chr10:393053-393076(-)</td>\n",
       "      <td>GCGATGTGTCCGGGTGAGTGTGG</td>\n",
       "      <td>GCG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>ATGTGTCCGGGTGAGTGTGG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           seqid  strand    chr     start           Sequence  \\\n",
       "729051  seq70792       0  chr10  37456123  GGTGCTCTGGAGACTAC   \n",
       "729052  seq70792       0  chr10  37467849  GGTGCTCTGGAGACTAC   \n",
       "729054  seq70793      16  chr17  80108904  GAACCTGGCGCACACCC   \n",
       "729055  seq70793      16  chr17  80107716  GAACCTGGCGCACACCC   \n",
       "729056  seq70793      16  chr17  80107752  GAACCTGGCGCACACCC   \n",
       "729057  seq70793      16  chr17  80109354  GAACCTGGCGCACACCC   \n",
       "729058  seq70794      16  chr10    393014  CACTCACCCGGACACAT   \n",
       "729059  seq70794      16  chr10    393421  CACTCACCCGGACACAT   \n",
       "729060  seq70794      16  chr10    393100  CACTCACCCGGACACAT   \n",
       "729061  seq70794      16  chr10    393057  CACTCACCCGGACACAT   \n",
       "\n",
       "                             chr_p                    seq23 NGG5 NGG3  \\\n",
       "729051  chr10:37456119-37456142(+)  GTGGGTGCTCTGGAGACTACTGG  GTG  TGG   \n",
       "729052  chr10:37467845-37467868(+)  GTGGGTGCTCTGGAGACTACTGG  GTG  TGG   \n",
       "729054  chr17:80108900-80108923(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG   \n",
       "729055  chr17:80107712-80107735(-)  TCCGGGTGTGCGCCAGGTTCTGG  TCC  TGG   \n",
       "729056  chr17:80107748-80107771(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG   \n",
       "729057  chr17:80109350-80109373(-)  TCTGGGTGTGCGCCAGGTTCTGG  TCT  TGG   \n",
       "729058      chr10:393010-393033(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG   \n",
       "729059      chr10:393417-393440(-)  GCAATGTGTCCGGGTGAGTGTGG  GCA  TGG   \n",
       "729060      chr10:393096-393119(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG   \n",
       "729061      chr10:393053-393076(-)  GCGATGTGTCCGGGTGAGTGTGG  GCG  TGG   \n",
       "\n",
       "                       seq20  \n",
       "729051  GGTGCTCTGGAGACTACTGG  \n",
       "729052  GGTGCTCTGGAGACTACTGG  \n",
       "729054  GGGTGTGCGCCAGGTTCTGG  \n",
       "729055  GGGTGTGCGCCAGGTTCTGG  \n",
       "729056  GGGTGTGCGCCAGGTTCTGG  \n",
       "729057  GGGTGTGCGCCAGGTTCTGG  \n",
       "729058  ATGTGTCCGGGTGAGTGTGG  \n",
       "729059  ATGTGTCCGGGTGAGTGTGG  \n",
       "729060  ATGTGTCCGGGTGAGTGTGG  \n",
       "729061  ATGTGTCCGGGTGAGTGTGG  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_alignments.tail(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Below is the final file. The data in this is not grouped, and hence this file can be used for further statistical analysis and plotting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "true_alignments.to_csv('/home/user/Desktop/nt_17/true_all_cols_17.csv',header=True,index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below I group the data to create a table that is easier to read, however this will make it difficult for analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>seq20</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr_p</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">chr1</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">seq1002</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">TACGGGGACTCCGTGGG</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">TACGGGGACTCCGTGGGGGG</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">0</th>\n",
       "      <th>chr1:1011583-1011606(+)</th>\n",
       "      <td>1011587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011613-1011636(+)</th>\n",
       "      <td>1011617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011643-1011666(+)</th>\n",
       "      <td>1011647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011673-1011696(+)</th>\n",
       "      <td>1011677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011881-1011904(+)</th>\n",
       "      <td>1011885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011911-1011934(+)</th>\n",
       "      <td>1011915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011941-1011964(+)</th>\n",
       "      <td>1011945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011971-1011994(+)</th>\n",
       "      <td>1011975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012001-1012024(+)</th>\n",
       "      <td>1012005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012031-1012054(+)</th>\n",
       "      <td>1012035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012061-1012084(+)</th>\n",
       "      <td>1012065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012091-1012114(+)</th>\n",
       "      <td>1012095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012121-1012144(+)</th>\n",
       "      <td>1012125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012151-1012174(+)</th>\n",
       "      <td>1012155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012181-1012204(+)</th>\n",
       "      <td>1012185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012211-1012234(+)</th>\n",
       "      <td>1012215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012241-1012264(+)</th>\n",
       "      <td>1012245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012271-1012294(+)</th>\n",
       "      <td>1012275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012301-1012324(+)</th>\n",
       "      <td>1012305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012331-1012354(+)</th>\n",
       "      <td>1012335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012361-1012384(+)</th>\n",
       "      <td>1012365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012391-1012414(+)</th>\n",
       "      <td>1012395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012421-1012444(+)</th>\n",
       "      <td>1012425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012751-1012774(+)</th>\n",
       "      <td>1012755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012781-1012804(+)</th>\n",
       "      <td>1012785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012811-1012834(+)</th>\n",
       "      <td>1012815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013111-1013134(+)</th>\n",
       "      <td>1013115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013141-1013164(+)</th>\n",
       "      <td>1013145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013171-1013194(+)</th>\n",
       "      <td>1013175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013261-1013284(+)</th>\n",
       "      <td>1013265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">chrY</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">seq9758</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAA</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAATGG</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">0</th>\n",
       "      <th>chrY:13718878-13718901(+)</th>\n",
       "      <td>13718882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13724485-13724508(+)</th>\n",
       "      <td>13724489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730087-13730110(+)</th>\n",
       "      <td>13730091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13735693-13735716(+)</th>\n",
       "      <td>13735697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13741300-13741323(+)</th>\n",
       "      <td>13741304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13746902-13746925(+)</th>\n",
       "      <td>13746906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9899</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690860-13690883(+)</th>\n",
       "      <td>13690864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696426-13696449(+)</th>\n",
       "      <td>13696430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699787-13699810(+)</th>\n",
       "      <td>13699791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13703148-13703171(+)</th>\n",
       "      <td>13703152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708772-13708795(+)</th>\n",
       "      <td>13708776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714423-13714446(+)</th>\n",
       "      <td>13714427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13720030-13720053(+)</th>\n",
       "      <td>13720034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725637-13725660(+)</th>\n",
       "      <td>13725641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13731239-13731262(+)</th>\n",
       "      <td>13731243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736845-13736868(+)</th>\n",
       "      <td>13736849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742452-13742475(+)</th>\n",
       "      <td>13742456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13748054-13748077(+)</th>\n",
       "      <td>13748058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9964</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690511-13690534(+)</th>\n",
       "      <td>13690515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696082-13696105(+)</th>\n",
       "      <td>13696086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699443-13699466(+)</th>\n",
       "      <td>13699447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13702804-13702827(+)</th>\n",
       "      <td>13702808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708428-13708451(+)</th>\n",
       "      <td>13708432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714079-13714102(+)</th>\n",
       "      <td>13714083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13719686-13719709(+)</th>\n",
       "      <td>13719690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725293-13725316(+)</th>\n",
       "      <td>13725297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730895-13730918(+)</th>\n",
       "      <td>13730899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736501-13736524(+)</th>\n",
       "      <td>13736505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742108-13742131(+)</th>\n",
       "      <td>13742112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13747710-13747733(+)</th>\n",
       "      <td>13747714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>554634 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                         start\n",
       "chr  seqid   Sequence          seq20                strand chr_p                              \n",
       "chr1 seq1002 TACGGGGACTCCGTGGG TACGGGGACTCCGTGGGGGG 0      chr1:1011583-1011606(+)     1011587\n",
       "                                                           chr1:1011613-1011636(+)     1011617\n",
       "                                                           chr1:1011643-1011666(+)     1011647\n",
       "                                                           chr1:1011673-1011696(+)     1011677\n",
       "                                                           chr1:1011881-1011904(+)     1011885\n",
       "                                                           chr1:1011911-1011934(+)     1011915\n",
       "                                                           chr1:1011941-1011964(+)     1011945\n",
       "                                                           chr1:1011971-1011994(+)     1011975\n",
       "                                                           chr1:1012001-1012024(+)     1012005\n",
       "                                                           chr1:1012031-1012054(+)     1012035\n",
       "                                                           chr1:1012061-1012084(+)     1012065\n",
       "                                                           chr1:1012091-1012114(+)     1012095\n",
       "                                                           chr1:1012121-1012144(+)     1012125\n",
       "                                                           chr1:1012151-1012174(+)     1012155\n",
       "                                                           chr1:1012181-1012204(+)     1012185\n",
       "                                                           chr1:1012211-1012234(+)     1012215\n",
       "                                                           chr1:1012241-1012264(+)     1012245\n",
       "                                                           chr1:1012271-1012294(+)     1012275\n",
       "                                                           chr1:1012301-1012324(+)     1012305\n",
       "                                                           chr1:1012331-1012354(+)     1012335\n",
       "                                                           chr1:1012361-1012384(+)     1012365\n",
       "                                                           chr1:1012391-1012414(+)     1012395\n",
       "                                                           chr1:1012421-1012444(+)     1012425\n",
       "                                                           chr1:1012751-1012774(+)     1012755\n",
       "                                                           chr1:1012781-1012804(+)     1012785\n",
       "                                                           chr1:1012811-1012834(+)     1012815\n",
       "                                                           chr1:1013111-1013134(+)     1013115\n",
       "                                                           chr1:1013141-1013164(+)     1013145\n",
       "                                                           chr1:1013171-1013194(+)     1013175\n",
       "                                                           chr1:1013261-1013284(+)     1013265\n",
       "...                                                                                        ...\n",
       "chrY seq9758 AATCAAATGGACAGGAA AATCAAATGGACAGGAATGG 0      chrY:13718878-13718901(+)  13718882\n",
       "                                                           chrY:13724485-13724508(+)  13724489\n",
       "                                                           chrY:13730087-13730110(+)  13730091\n",
       "                                                           chrY:13735693-13735716(+)  13735697\n",
       "                                                           chrY:13741300-13741323(+)  13741304\n",
       "                                                           chrY:13746902-13746925(+)  13746906\n",
       "     seq9899 AATGCAACGGAGTGGAA AATGCAACGGAGTGGAATGG 0      chrY:13690860-13690883(+)  13690864\n",
       "                                                           chrY:13696426-13696449(+)  13696430\n",
       "                                                           chrY:13699787-13699810(+)  13699791\n",
       "                                                           chrY:13703148-13703171(+)  13703152\n",
       "                                                           chrY:13708772-13708795(+)  13708776\n",
       "                                                           chrY:13714423-13714446(+)  13714427\n",
       "                                                           chrY:13720030-13720053(+)  13720034\n",
       "                                                           chrY:13725637-13725660(+)  13725641\n",
       "                                                           chrY:13731239-13731262(+)  13731243\n",
       "                                                           chrY:13736845-13736868(+)  13736849\n",
       "                                                           chrY:13742452-13742475(+)  13742456\n",
       "                                                           chrY:13748054-13748077(+)  13748058\n",
       "     seq9964 GATGGAATGGCGTGTAA GATGGAATGGCGTGTAATGG 0      chrY:13690511-13690534(+)  13690515\n",
       "                                                           chrY:13696082-13696105(+)  13696086\n",
       "                                                           chrY:13699443-13699466(+)  13699447\n",
       "                                                           chrY:13702804-13702827(+)  13702808\n",
       "                                                           chrY:13708428-13708451(+)  13708432\n",
       "                                                           chrY:13714079-13714102(+)  13714083\n",
       "                                                           chrY:13719686-13719709(+)  13719690\n",
       "                                                           chrY:13725293-13725316(+)  13725297\n",
       "                                                           chrY:13730895-13730918(+)  13730899\n",
       "                                                           chrY:13736501-13736524(+)  13736505\n",
       "                                                           chrY:13742108-13742131(+)  13742112\n",
       "                                                           chrY:13747710-13747733(+)  13747714\n",
       "\n",
       "[554634 rows x 1 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#creating pivot table\n",
    "true_alignments.pivot_table(index=['chr','seqid','Sequence','seq20','strand','chr_p'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "coun_temp = true_alignments.pivot_table(index=['chr','seqid'],aggfunc=[len])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "coun_temp.sortlevel(1,inplace=True)\n",
    "coun_temp.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This output below is fine except it would be nice to have another column that has a count of the number of times the sequences is repeated."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>seqid</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>seq20</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr_p</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"20\" valign=\"top\">chr1</th>\n",
       "      <th rowspan=\"20\" valign=\"top\">seq1002</th>\n",
       "      <th rowspan=\"20\" valign=\"top\">TACGGGGACTCCGTGGG</th>\n",
       "      <th rowspan=\"20\" valign=\"top\">TACGGGGACTCCGTGGGGGG</th>\n",
       "      <th rowspan=\"20\" valign=\"top\">0</th>\n",
       "      <th>chr1:1011583-1011606(+)</th>\n",
       "      <td>1011587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011613-1011636(+)</th>\n",
       "      <td>1011617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011643-1011666(+)</th>\n",
       "      <td>1011647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011673-1011696(+)</th>\n",
       "      <td>1011677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011881-1011904(+)</th>\n",
       "      <td>1011885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011911-1011934(+)</th>\n",
       "      <td>1011915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011941-1011964(+)</th>\n",
       "      <td>1011945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011971-1011994(+)</th>\n",
       "      <td>1011975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012001-1012024(+)</th>\n",
       "      <td>1012005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012031-1012054(+)</th>\n",
       "      <td>1012035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012061-1012084(+)</th>\n",
       "      <td>1012065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012091-1012114(+)</th>\n",
       "      <td>1012095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012121-1012144(+)</th>\n",
       "      <td>1012125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012151-1012174(+)</th>\n",
       "      <td>1012155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012181-1012204(+)</th>\n",
       "      <td>1012185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012211-1012234(+)</th>\n",
       "      <td>1012215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012241-1012264(+)</th>\n",
       "      <td>1012245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012271-1012294(+)</th>\n",
       "      <td>1012275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012301-1012324(+)</th>\n",
       "      <td>1012305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012331-1012354(+)</th>\n",
       "      <td>1012335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                      start\n",
       "chr  seqid   Sequence          seq20                strand chr_p                           \n",
       "chr1 seq1002 TACGGGGACTCCGTGGG TACGGGGACTCCGTGGGGGG 0      chr1:1011583-1011606(+)  1011587\n",
       "                                                           chr1:1011613-1011636(+)  1011617\n",
       "                                                           chr1:1011643-1011666(+)  1011647\n",
       "                                                           chr1:1011673-1011696(+)  1011677\n",
       "                                                           chr1:1011881-1011904(+)  1011885\n",
       "                                                           chr1:1011911-1011934(+)  1011915\n",
       "                                                           chr1:1011941-1011964(+)  1011945\n",
       "                                                           chr1:1011971-1011994(+)  1011975\n",
       "                                                           chr1:1012001-1012024(+)  1012005\n",
       "                                                           chr1:1012031-1012054(+)  1012035\n",
       "                                                           chr1:1012061-1012084(+)  1012065\n",
       "                                                           chr1:1012091-1012114(+)  1012095\n",
       "                                                           chr1:1012121-1012144(+)  1012125\n",
       "                                                           chr1:1012151-1012174(+)  1012155\n",
       "                                                           chr1:1012181-1012204(+)  1012185\n",
       "                                                           chr1:1012211-1012234(+)  1012215\n",
       "                                                           chr1:1012241-1012264(+)  1012245\n",
       "                                                           chr1:1012271-1012294(+)  1012275\n",
       "                                                           chr1:1012301-1012324(+)  1012305\n",
       "                                                           chr1:1012331-1012354(+)  1012335"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_alignments_table = true_alignments.pivot_table(index=['chr','seqid','Sequence','seq20','strand','chr_p'])\n",
    "true_alignments_table.head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the cells below I modefiy the multi-index in panda to include a column for the counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chr10:1000689-1000712(-)', 'chr10:1000690-1000713(-)',\n",
       "       'chr10:1000706-1000729(-)', 'chr10:1000715-1000738(-)',\n",
       "       'chr10:1000716-1000739(-)', 'chr10:1000732-1000755(-)',\n",
       "       'chr10:1000741-1000764(-)', 'chr10:1000742-1000765(-)',\n",
       "       'chr10:1000758-1000781(-)', 'chr10:1000767-1000790(-)',\n",
       "       ...\n",
       "       'chrY:58917620-58917643(-)', 'chrY:58917636-58917659(-)',\n",
       "       'chrY:58993054-58993077(+)', 'chrY:58993055-58993078(+)',\n",
       "       'chrY:58995233-58995256(+)', 'chrY:58996054-58996077(+)',\n",
       "       'chrY:58996055-58996078(+)', 'chrY:58996122-58996145(+)',\n",
       "       'chrY:58996123-58996146(+)', 'chrY:58996739-58996762(+)'],\n",
       "      dtype='object', name='chr_p', length=554634)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_alignments_table.index.levels[5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FrozenNDArray([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...], dtype='int8')"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_alignments_table.index.labels[4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "ta1= true_alignments_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "levels_ta1=[ta1.index.levels[0],ta1.index.levels[1],[coun_temp.len.NGG3.values],ta1.index.levels[2],ta1.index.levels[3],ta1.index.levels[4],ta1.index.levels[5]]\n",
    "labels_ta1 = [ta1.index.labels[0],ta1.index.labels[1],ta1.index.labels[1],ta1.index.labels[2],ta1.index.labels[3],ta1.index.labels[4],ta1.index.labels[5]]\n",
    "\n",
    "ta1_index = pd.MultiIndex(levels=levels_ta1,labels=labels_ta1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "ta1.set_index(ta1_index,inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>seqid</th>\n",
       "      <th></th>\n",
       "      <th>Sequence</th>\n",
       "      <th>seq20</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr_p</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"61\" valign=\"top\">chrY</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">seq55600</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">GACGCAGCAGACCTTTA</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">GACGCAGCAGACCTTTACGG</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">0</th>\n",
       "      <th>chrY:22460080-22460103(+)</th>\n",
       "      <td>22460084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22466805-22466828(+)</th>\n",
       "      <td>22466809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22471164-22471187(+)</th>\n",
       "      <td>22471168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq55678</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">AGTGCCCAGGAGTGGAA</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">AGTGCCCAGGAGTGGAATGG</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">0</th>\n",
       "      <th>chrY:28795346-28795369(+)</th>\n",
       "      <td>28795350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:28804733-28804756(+)</th>\n",
       "      <td>28804737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:28813322-28813345(+)</th>\n",
       "      <td>28813326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:28817158-28817181(+)</th>\n",
       "      <td>28817162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq55697</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">AGTAAACGGACAAGAGA</th>\n",
       "      <th>AGTAAACGGACAAGAGAAGG</th>\n",
       "      <th>0</th>\n",
       "      <th>chrY:20870976-20870999(+)</th>\n",
       "      <td>20870980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGTAAACGGACAAGAGAGGG</th>\n",
       "      <th>0</th>\n",
       "      <th>chrY:20861151-20861174(+)</th>\n",
       "      <td>20861155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">TCTCTTGTCCGTTTACT</th>\n",
       "      <th>AGTAAACGGACAAGAGAAGG</th>\n",
       "      <th>16</th>\n",
       "      <th>chrY:20773174-20773197(-)</th>\n",
       "      <td>20773178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGTAAACGGACAAGAGAGGG</th>\n",
       "      <th>16</th>\n",
       "      <th>chrY:20782997-20783020(-)</th>\n",
       "      <td>20783001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq55864</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th>CCAAGGTTTACAGCTTC</th>\n",
       "      <th>GAAGCTGTAAACCTTGGTGG</th>\n",
       "      <th>16</th>\n",
       "      <th>chrY:22236089-22236112(-)</th>\n",
       "      <td>22236093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">GAAGCTGTAAACCTTGG</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">GAAGCTGTAAACCTTGGTGG</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">0</th>\n",
       "      <th>chrY:22256202-22256225(+)</th>\n",
       "      <td>22256206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22273045-22273068(+)</th>\n",
       "      <td>22273049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22273170-22273193(+)</th>\n",
       "      <td>22273174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq55866</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">TGAAGCTGCAGACCGTT</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">TGAAGCTGCAGACCGTTAGG</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">0</th>\n",
       "      <th>chrY:22269526-22269549(+)</th>\n",
       "      <td>22269530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22274170-22274193(+)</th>\n",
       "      <td>22274174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22316771-22316794(+)</th>\n",
       "      <td>22316775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:22353557-22353580(+)</th>\n",
       "      <td>22353561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq55893</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">AAAAATGATGGCTTCAG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">CTGAAGCCATCATTTTTTGG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">16</th>\n",
       "      <th>chrY:20774165-20774188(-)</th>\n",
       "      <td>20774169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:20783990-20784013(-)</th>\n",
       "      <td>20783994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">CTGAAGCCATCATTTTT</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">CTGAAGCCATCATTTTTTGG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">0</th>\n",
       "      <th>chrY:20860158-20860181(+)</th>\n",
       "      <td>20860162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:20869985-20870008(+)</th>\n",
       "      <td>20869989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">seq56277</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">4</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">ACAGATTTGCTTCCTCT</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">ACAGATTTGCTTCCTCTGGG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">0</th>\n",
       "      <th>chrY:20860386-20860409(+)</th>\n",
       "      <td>20860390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:20870212-20870235(+)</th>\n",
       "      <td>20870216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">AGAGGAAGCAAATCTGT</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">ACAGATTTGCTTCCTCTGGG</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">16</th>\n",
       "      <th>chrY:20773938-20773961(-)</th>\n",
       "      <td>20773942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:20783762-20783785(-)</th>\n",
       "      <td>20783766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">seq5661</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">16</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">GTCTCTTCACTTCGATT</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">AATCGAAGTGAAGAGACTGG</th>\n",
       "      <th rowspan=\"3\" valign=\"top\">16</th>\n",
       "      <th>chrY:58825848-58825871(-)</th>\n",
       "      <td>58825852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:58829412-58829435(-)</th>\n",
       "      <td>58829416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:58833001-58833024(-)</th>\n",
       "      <td>58833005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>seq5661</th>\n",
       "      <th>16</th>\n",
       "      <th>GTCTCTTCACTTCGATT</th>\n",
       "      <th>AATCGAAGTGAAGAGACTGG</th>\n",
       "      <th>16</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">seq9758</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAA</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAATGG</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">0</th>\n",
       "      <th>chrY:13718878-13718901(+)</th>\n",
       "      <td>13718882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13724485-13724508(+)</th>\n",
       "      <td>13724489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730087-13730110(+)</th>\n",
       "      <td>13730091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13735693-13735716(+)</th>\n",
       "      <td>13735697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13741300-13741323(+)</th>\n",
       "      <td>13741304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13746902-13746925(+)</th>\n",
       "      <td>13746906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9899</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690860-13690883(+)</th>\n",
       "      <td>13690864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696426-13696449(+)</th>\n",
       "      <td>13696430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699787-13699810(+)</th>\n",
       "      <td>13699791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13703148-13703171(+)</th>\n",
       "      <td>13703152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708772-13708795(+)</th>\n",
       "      <td>13708776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714423-13714446(+)</th>\n",
       "      <td>13714427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13720030-13720053(+)</th>\n",
       "      <td>13720034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725637-13725660(+)</th>\n",
       "      <td>13725641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13731239-13731262(+)</th>\n",
       "      <td>13731243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736845-13736868(+)</th>\n",
       "      <td>13736849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742452-13742475(+)</th>\n",
       "      <td>13742456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13748054-13748077(+)</th>\n",
       "      <td>13748058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9964</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690511-13690534(+)</th>\n",
       "      <td>13690515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696082-13696105(+)</th>\n",
       "      <td>13696086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699443-13699466(+)</th>\n",
       "      <td>13699447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13702804-13702827(+)</th>\n",
       "      <td>13702808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708428-13708451(+)</th>\n",
       "      <td>13708432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714079-13714102(+)</th>\n",
       "      <td>13714083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13719686-13719709(+)</th>\n",
       "      <td>13719690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725293-13725316(+)</th>\n",
       "      <td>13725297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730895-13730918(+)</th>\n",
       "      <td>13730899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736501-13736524(+)</th>\n",
       "      <td>13736505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742108-13742131(+)</th>\n",
       "      <td>13742112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13747710-13747733(+)</th>\n",
       "      <td>13747714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>800 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                             start\n",
       "chr  seqid       Sequence          seq20                strand chr_p                              \n",
       "chrY seq55600 4  GACGCAGCAGACCTTTA GACGCAGCAGACCTTTACGG 0      chrY:22460080-22460103(+)  22460084\n",
       "                                                               chrY:22466805-22466828(+)  22466809\n",
       "                                                               chrY:22471164-22471187(+)  22471168\n",
       "     seq55678 4  AGTGCCCAGGAGTGGAA AGTGCCCAGGAGTGGAATGG 0      chrY:28795346-28795369(+)  28795350\n",
       "                                                               chrY:28804733-28804756(+)  28804737\n",
       "                                                               chrY:28813322-28813345(+)  28813326\n",
       "                                                               chrY:28817158-28817181(+)  28817162\n",
       "     seq55697 4  AGTAAACGGACAAGAGA AGTAAACGGACAAGAGAAGG 0      chrY:20870976-20870999(+)  20870980\n",
       "                                   AGTAAACGGACAAGAGAGGG 0      chrY:20861151-20861174(+)  20861155\n",
       "                 TCTCTTGTCCGTTTACT AGTAAACGGACAAGAGAAGG 16     chrY:20773174-20773197(-)  20773178\n",
       "                                   AGTAAACGGACAAGAGAGGG 16     chrY:20782997-20783020(-)  20783001\n",
       "     seq55864 4  CCAAGGTTTACAGCTTC GAAGCTGTAAACCTTGGTGG 16     chrY:22236089-22236112(-)  22236093\n",
       "                 GAAGCTGTAAACCTTGG GAAGCTGTAAACCTTGGTGG 0      chrY:22256202-22256225(+)  22256206\n",
       "                                                               chrY:22273045-22273068(+)  22273049\n",
       "                                                               chrY:22273170-22273193(+)  22273174\n",
       "     seq55866 4  TGAAGCTGCAGACCGTT TGAAGCTGCAGACCGTTAGG 0      chrY:22269526-22269549(+)  22269530\n",
       "                                                               chrY:22274170-22274193(+)  22274174\n",
       "                                                               chrY:22316771-22316794(+)  22316775\n",
       "                                                               chrY:22353557-22353580(+)  22353561\n",
       "     seq55893 4  AAAAATGATGGCTTCAG CTGAAGCCATCATTTTTTGG 16     chrY:20774165-20774188(-)  20774169\n",
       "                                                               chrY:20783990-20784013(-)  20783994\n",
       "                 CTGAAGCCATCATTTTT CTGAAGCCATCATTTTTTGG 0      chrY:20860158-20860181(+)  20860162\n",
       "                                                               chrY:20869985-20870008(+)  20869989\n",
       "     seq56277 4  ACAGATTTGCTTCCTCT ACAGATTTGCTTCCTCTGGG 0      chrY:20860386-20860409(+)  20860390\n",
       "                                                               chrY:20870212-20870235(+)  20870216\n",
       "                 AGAGGAAGCAAATCTGT ACAGATTTGCTTCCTCTGGG 16     chrY:20773938-20773961(-)  20773942\n",
       "                                                               chrY:20783762-20783785(-)  20783766\n",
       "     seq5661  16 GTCTCTTCACTTCGATT AATCGAAGTGAAGAGACTGG 16     chrY:58825848-58825871(-)  58825852\n",
       "                                                               chrY:58829412-58829435(-)  58829416\n",
       "                                                               chrY:58833001-58833024(-)  58833005\n",
       "...                                                                                            ...\n",
       "     seq9758  12 AATCAAATGGACAGGAA AATCAAATGGACAGGAATGG 0      chrY:13718878-13718901(+)  13718882\n",
       "                                                               chrY:13724485-13724508(+)  13724489\n",
       "                                                               chrY:13730087-13730110(+)  13730091\n",
       "                                                               chrY:13735693-13735716(+)  13735697\n",
       "                                                               chrY:13741300-13741323(+)  13741304\n",
       "                                                               chrY:13746902-13746925(+)  13746906\n",
       "     seq9899  12 AATGCAACGGAGTGGAA AATGCAACGGAGTGGAATGG 0      chrY:13690860-13690883(+)  13690864\n",
       "                                                               chrY:13696426-13696449(+)  13696430\n",
       "                                                               chrY:13699787-13699810(+)  13699791\n",
       "                                                               chrY:13703148-13703171(+)  13703152\n",
       "                                                               chrY:13708772-13708795(+)  13708776\n",
       "                                                               chrY:13714423-13714446(+)  13714427\n",
       "                                                               chrY:13720030-13720053(+)  13720034\n",
       "                                                               chrY:13725637-13725660(+)  13725641\n",
       "                                                               chrY:13731239-13731262(+)  13731243\n",
       "                                                               chrY:13736845-13736868(+)  13736849\n",
       "                                                               chrY:13742452-13742475(+)  13742456\n",
       "                                                               chrY:13748054-13748077(+)  13748058\n",
       "     seq9964  12 GATGGAATGGCGTGTAA GATGGAATGGCGTGTAATGG 0      chrY:13690511-13690534(+)  13690515\n",
       "                                                               chrY:13696082-13696105(+)  13696086\n",
       "                                                               chrY:13699443-13699466(+)  13699447\n",
       "                                                               chrY:13702804-13702827(+)  13702808\n",
       "                                                               chrY:13708428-13708451(+)  13708432\n",
       "                                                               chrY:13714079-13714102(+)  13714083\n",
       "                                                               chrY:13719686-13719709(+)  13719690\n",
       "                                                               chrY:13725293-13725316(+)  13725297\n",
       "                                                               chrY:13730895-13730918(+)  13730899\n",
       "                                                               chrY:13736501-13736524(+)  13736505\n",
       "                                                               chrY:13742108-13742131(+)  13742112\n",
       "                                                               chrY:13747710-13747733(+)  13747714\n",
       "\n",
       "[800 rows x 1 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ta1.tail(800)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "ta1_names = ['chr', 'seqid', 'count', 'Sequence', 'seq20', 'strand', 'chr_p']\n",
    "ta1.index.names = ta1_names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Below is the final output with the counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>seqid</th>\n",
       "      <th>count</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>seq20</th>\n",
       "      <th>strand</th>\n",
       "      <th>chr_p</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">chr1</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">seq1002</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">33</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">TACGGGGACTCCGTGGG</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">TACGGGGACTCCGTGGGGGG</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">0</th>\n",
       "      <th>chr1:1011583-1011606(+)</th>\n",
       "      <td>1011587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011613-1011636(+)</th>\n",
       "      <td>1011617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011643-1011666(+)</th>\n",
       "      <td>1011647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011673-1011696(+)</th>\n",
       "      <td>1011677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011881-1011904(+)</th>\n",
       "      <td>1011885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011911-1011934(+)</th>\n",
       "      <td>1011915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011941-1011964(+)</th>\n",
       "      <td>1011945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1011971-1011994(+)</th>\n",
       "      <td>1011975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012001-1012024(+)</th>\n",
       "      <td>1012005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012031-1012054(+)</th>\n",
       "      <td>1012035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012061-1012084(+)</th>\n",
       "      <td>1012065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012091-1012114(+)</th>\n",
       "      <td>1012095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012121-1012144(+)</th>\n",
       "      <td>1012125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012151-1012174(+)</th>\n",
       "      <td>1012155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012181-1012204(+)</th>\n",
       "      <td>1012185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012211-1012234(+)</th>\n",
       "      <td>1012215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012241-1012264(+)</th>\n",
       "      <td>1012245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012271-1012294(+)</th>\n",
       "      <td>1012275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012301-1012324(+)</th>\n",
       "      <td>1012305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012331-1012354(+)</th>\n",
       "      <td>1012335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012361-1012384(+)</th>\n",
       "      <td>1012365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012391-1012414(+)</th>\n",
       "      <td>1012395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012421-1012444(+)</th>\n",
       "      <td>1012425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012751-1012774(+)</th>\n",
       "      <td>1012755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012781-1012804(+)</th>\n",
       "      <td>1012785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1012811-1012834(+)</th>\n",
       "      <td>1012815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013111-1013134(+)</th>\n",
       "      <td>1013115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013141-1013164(+)</th>\n",
       "      <td>1013145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013171-1013194(+)</th>\n",
       "      <td>1013175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1013261-1013284(+)</th>\n",
       "      <td>1013265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">chrY</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">seq9758</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAA</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">AATCAAATGGACAGGAATGG</th>\n",
       "      <th rowspan=\"6\" valign=\"top\">0</th>\n",
       "      <th>chrY:13718878-13718901(+)</th>\n",
       "      <td>13718882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13724485-13724508(+)</th>\n",
       "      <td>13724489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730087-13730110(+)</th>\n",
       "      <td>13730091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13735693-13735716(+)</th>\n",
       "      <td>13735697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13741300-13741323(+)</th>\n",
       "      <td>13741304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13746902-13746925(+)</th>\n",
       "      <td>13746906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9899</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">AATGCAACGGAGTGGAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690860-13690883(+)</th>\n",
       "      <td>13690864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696426-13696449(+)</th>\n",
       "      <td>13696430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699787-13699810(+)</th>\n",
       "      <td>13699791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13703148-13703171(+)</th>\n",
       "      <td>13703152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708772-13708795(+)</th>\n",
       "      <td>13708776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714423-13714446(+)</th>\n",
       "      <td>13714427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13720030-13720053(+)</th>\n",
       "      <td>13720034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725637-13725660(+)</th>\n",
       "      <td>13725641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13731239-13731262(+)</th>\n",
       "      <td>13731243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736845-13736868(+)</th>\n",
       "      <td>13736849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742452-13742475(+)</th>\n",
       "      <td>13742456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13748054-13748077(+)</th>\n",
       "      <td>13748058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"12\" valign=\"top\">seq9964</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">12</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAA</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">GATGGAATGGCGTGTAATGG</th>\n",
       "      <th rowspan=\"12\" valign=\"top\">0</th>\n",
       "      <th>chrY:13690511-13690534(+)</th>\n",
       "      <td>13690515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13696082-13696105(+)</th>\n",
       "      <td>13696086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13699443-13699466(+)</th>\n",
       "      <td>13699447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13702804-13702827(+)</th>\n",
       "      <td>13702808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13708428-13708451(+)</th>\n",
       "      <td>13708432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13714079-13714102(+)</th>\n",
       "      <td>13714083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13719686-13719709(+)</th>\n",
       "      <td>13719690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13725293-13725316(+)</th>\n",
       "      <td>13725297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13730895-13730918(+)</th>\n",
       "      <td>13730899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13736501-13736524(+)</th>\n",
       "      <td>13736505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13742108-13742131(+)</th>\n",
       "      <td>13742112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY:13747710-13747733(+)</th>\n",
       "      <td>13747714</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>554634 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                               start\n",
       "chr  seqid   count Sequence          seq20                strand chr_p                              \n",
       "chr1 seq1002 33    TACGGGGACTCCGTGGG TACGGGGACTCCGTGGGGGG 0      chr1:1011583-1011606(+)     1011587\n",
       "                                                                 chr1:1011613-1011636(+)     1011617\n",
       "                                                                 chr1:1011643-1011666(+)     1011647\n",
       "                                                                 chr1:1011673-1011696(+)     1011677\n",
       "                                                                 chr1:1011881-1011904(+)     1011885\n",
       "                                                                 chr1:1011911-1011934(+)     1011915\n",
       "                                                                 chr1:1011941-1011964(+)     1011945\n",
       "                                                                 chr1:1011971-1011994(+)     1011975\n",
       "                                                                 chr1:1012001-1012024(+)     1012005\n",
       "                                                                 chr1:1012031-1012054(+)     1012035\n",
       "                                                                 chr1:1012061-1012084(+)     1012065\n",
       "                                                                 chr1:1012091-1012114(+)     1012095\n",
       "                                                                 chr1:1012121-1012144(+)     1012125\n",
       "                                                                 chr1:1012151-1012174(+)     1012155\n",
       "                                                                 chr1:1012181-1012204(+)     1012185\n",
       "                                                                 chr1:1012211-1012234(+)     1012215\n",
       "                                                                 chr1:1012241-1012264(+)     1012245\n",
       "                                                                 chr1:1012271-1012294(+)     1012275\n",
       "                                                                 chr1:1012301-1012324(+)     1012305\n",
       "                                                                 chr1:1012331-1012354(+)     1012335\n",
       "                                                                 chr1:1012361-1012384(+)     1012365\n",
       "                                                                 chr1:1012391-1012414(+)     1012395\n",
       "                                                                 chr1:1012421-1012444(+)     1012425\n",
       "                                                                 chr1:1012751-1012774(+)     1012755\n",
       "                                                                 chr1:1012781-1012804(+)     1012785\n",
       "                                                                 chr1:1012811-1012834(+)     1012815\n",
       "                                                                 chr1:1013111-1013134(+)     1013115\n",
       "                                                                 chr1:1013141-1013164(+)     1013145\n",
       "                                                                 chr1:1013171-1013194(+)     1013175\n",
       "                                                                 chr1:1013261-1013284(+)     1013265\n",
       "...                                                                                              ...\n",
       "chrY seq9758 12    AATCAAATGGACAGGAA AATCAAATGGACAGGAATGG 0      chrY:13718878-13718901(+)  13718882\n",
       "                                                                 chrY:13724485-13724508(+)  13724489\n",
       "                                                                 chrY:13730087-13730110(+)  13730091\n",
       "                                                                 chrY:13735693-13735716(+)  13735697\n",
       "                                                                 chrY:13741300-13741323(+)  13741304\n",
       "                                                                 chrY:13746902-13746925(+)  13746906\n",
       "     seq9899 12    AATGCAACGGAGTGGAA AATGCAACGGAGTGGAATGG 0      chrY:13690860-13690883(+)  13690864\n",
       "                                                                 chrY:13696426-13696449(+)  13696430\n",
       "                                                                 chrY:13699787-13699810(+)  13699791\n",
       "                                                                 chrY:13703148-13703171(+)  13703152\n",
       "                                                                 chrY:13708772-13708795(+)  13708776\n",
       "                                                                 chrY:13714423-13714446(+)  13714427\n",
       "                                                                 chrY:13720030-13720053(+)  13720034\n",
       "                                                                 chrY:13725637-13725660(+)  13725641\n",
       "                                                                 chrY:13731239-13731262(+)  13731243\n",
       "                                                                 chrY:13736845-13736868(+)  13736849\n",
       "                                                                 chrY:13742452-13742475(+)  13742456\n",
       "                                                                 chrY:13748054-13748077(+)  13748058\n",
       "     seq9964 12    GATGGAATGGCGTGTAA GATGGAATGGCGTGTAATGG 0      chrY:13690511-13690534(+)  13690515\n",
       "                                                                 chrY:13696082-13696105(+)  13696086\n",
       "                                                                 chrY:13699443-13699466(+)  13699447\n",
       "                                                                 chrY:13702804-13702827(+)  13702808\n",
       "                                                                 chrY:13708428-13708451(+)  13708432\n",
       "                                                                 chrY:13714079-13714102(+)  13714083\n",
       "                                                                 chrY:13719686-13719709(+)  13719690\n",
       "                                                                 chrY:13725293-13725316(+)  13725297\n",
       "                                                                 chrY:13730895-13730918(+)  13730899\n",
       "                                                                 chrY:13736501-13736524(+)  13736505\n",
       "                                                                 chrY:13742108-13742131(+)  13742112\n",
       "                                                                 chrY:13747710-13747733(+)  13747714\n",
       "\n",
       "[554634 rows x 1 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ta1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "ta1.to_excel('/home/user/Desktop/nt_17/true_grouped_17.xlsx',header=True,columns=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
